modin/pandas/groupby.py

```diff
@@ -834,28 +834,16 @@ def _apply_agg_function(self, f, drop=True, *args, **kwargs):
         """
         assert callable(f), "'{0}' object is not callable".format(type(f))
 
-        f = wrap_udf_function(f)
-        if self._is_multi_by:
-            return self._default_to_pandas(f, *args, **kwargs)
-
-        if isinstance(self._by, type(self._query_compiler)):
-            by = self._by.to_pandas().squeeze()
-        else:
-            by = self._by
-
-        # For aggregations, pandas behavior does this for the result.
-        # For other operations it does not, so we wait until there is an aggregation to
-        # actually perform this operation.
-        if self._idx_name is not None and drop and self._drop:
-            groupby_qc = self._query_compiler.drop(columns=[self._idx_name])
-        else:
-            groupby_qc = self._query_compiler
```

Comment:
why this logic was moved to backend?

Reply:
Because it depends on `is_multi_by` which is checked above.

Reply:
well, then we can just add `if not self._is_multi_by and ...` to that if statement?

Reply:
I'll try to move that logic back, and post about results (is it possible or not)

Reply:
Feel free to push these changes into this branch if everything is ok.

Reply:
done it